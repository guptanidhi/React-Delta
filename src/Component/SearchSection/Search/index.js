import React from 'react'
import PropTypes from 'prop-types'
import {InputGroup, FormControl, Button } from 'react-bootstrap';
import Grid from '../../Grid'

const Search = ({searchBtnText, searchPlaceHolder}) => {
  return (		
		<Grid>
			<InputGroup>
				<FormControl
					placeholder={searchPlaceHolder}
					aria-label="Press Enter To Search"
					aria-describedby="basic-addon2"
				/>
				<InputGroup.Append>
					<Button variant="outline-secondary">{searchBtnText}</Button>
				</InputGroup.Append>
			</InputGroup>
		</Grid>
	)
}

/** Props **/
Search.defaultProps = {
	searchPlaceHolder: 'Press Enter To Search',
	searchBtnText: 'search'
}

Search.propTypes = {
	searchPlaceHolder: PropTypes.string,
	searchBtnText: PropTypes.string
}

export default Search