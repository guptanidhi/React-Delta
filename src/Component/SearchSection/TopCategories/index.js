import React from 'react';
import PropTypes from 'prop-types'
import Grid from '../../Grid'
import {Button } from 'react-bootstrap';
import styles from './styles.module.scss';


const TopCategories = ({topCategories}) => {
   return (
    <div className={styles.requestSection}>
			<Grid>
				<div className={styles.requestBtn}>
						{topCategories.map((item)=>(
							<div id={item.id} className={styles.requestList}>
								<Button variant="dark">{item.name}</Button>
							</div>)
						)}
				</div>
			</Grid>
    </div>
   )
}

/** Props **/
TopCategories.defaultProps = {
	topCategories: []
}

TopCategories.propTypes = {
	topCategories: PropTypes.array
}

export default TopCategories;
