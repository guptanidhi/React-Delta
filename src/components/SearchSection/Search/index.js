import React from 'react'
import PropTypes from 'prop-types'
import {InputGroup, FormControl, Button } from 'react-bootstrap';
import style from './search.module.scss'
import Icon from '../../Icon'

const Search = ({searchBtnText, searchPlaceHolder}) => {
  return (		
			<InputGroup className={style.inputWrapper}>
				<FormControl
				    className={style.searchInput}
					placeholder={searchPlaceHolder}
					aria-label="Press Enter To Search"
					aria-describedby="basic-addon2"
				/>
				<InputGroup.Append>
					<Button variant="outline-secondary"> <Icon icon="filter"/> {searchBtnText}</Button>
				</InputGroup.Append>
			</InputGroup>
	)
}

/** Props **/
Search.defaultProps = {
	searchPlaceHolder: 'Press Enter To Search',
	searchBtnText: 'search'
}

Search.propTypes = {
	searchPlaceHolder: PropTypes.string,
	searchBtnText: PropTypes.string
}

export default Search