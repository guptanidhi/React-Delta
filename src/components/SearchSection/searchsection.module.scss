.headingSection{
	text-align: center;
	padding: 20px 0 30px;
}

.formSection, .requestSection, .headingSection {
  background-color: #f7f7f7;
}

.headingWrapper {
	margin-bottom: 15px;
}