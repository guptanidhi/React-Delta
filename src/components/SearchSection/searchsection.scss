.headingSection{
	text-align: center;
	padding: 20px 0;
}
.formSection, .requestSection, .headingSection {
  background-color: #f7f7f7;
}
