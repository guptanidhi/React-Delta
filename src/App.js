import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'slick-carousel/slick/slick.css'; 
import 'slick-carousel/slick/slick-theme.css';
import CustomNavbar from './components/Navbar';
import SearchSection from './components/SearchSection';
import styles from './app.module.scss';
import CustomSlider from './components/Slider'
import { topCategories, navItems, sliderSettings } from './jsonData/data';
import carouselData from './jsonData/carouselData'

const App = () => (
  <React.Fragment>    
    <CustomNavbar navItems={navItems} />
    <SearchSection
      title="Datasets"
      searchBtnText="Filter"
      topCategoriesData={topCategories} />
    <div className={styles.carouselSection}>
      <CustomSlider carouselTitle="Trending Datasets" slides={carouselData} sliderSettingObj={sliderSettings} />
    </div>
    <div className={styles.carouselSection}>
      <CustomSlider carouselTitle="Popular Datasets" slides={carouselData} sliderSettingObj={sliderSettings} />
    </div>
    <div className={styles.carouselSection}>
      <CustomSlider carouselTitle="Relevant Datasets" slides={carouselData} sliderSettingObj={sliderSettings} />
    </div>
  </React.Fragment>
);

export default App;
