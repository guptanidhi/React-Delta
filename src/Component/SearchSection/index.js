import React from 'react';
import PropTypes from 'prop-types'
import Grid from '../Grid'
import Heading from '../Heading'
import styles from './searchsection.scss';
import TopCategories from './TopCategories';
import Search from './Search'

const SearchSection = (props) => {
	const { title, searchBtnText, topCategoriesData } = props;
   return (
		<>
			<div className={styles.headingSection}>
				<Grid>
					<Heading type="h1">{title}</Heading>
				</Grid>
	 		</div>
			<div className={styles.formSection}>
				<Search searchBtnText={searchBtnText}/>
			</div>
			<TopCategories topCategories={topCategoriesData}/>
		</>
	 )
}

/** Props **/
SearchSection.defaultProps = {
	title: '',
	searchBtnText: 'search',
	topCategoriesData: []
}

SearchSection.propTypes = {
	title: PropTypes.string,
	searchBtnText: PropTypes.string,
	topCategoriesData: PropTypes.array
}

export default SearchSection;
