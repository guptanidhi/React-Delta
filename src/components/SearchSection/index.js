import React from 'react';
import PropTypes from 'prop-types'
import Grid from '../Grid'
import Heading from '../Heading'
import TopCategories from './TopCategories';
import Search from './Search'
import styles from './searchsection.module.scss'

const SearchSection = (props) => {
	const { title, searchBtnText, topCategoriesData } = props;
   return (
		<React.Fragment>
			<div className={styles.headingSection}>
				<Grid>
					<div className={styles.alignSection}>
					<div className={styles.headingWrapper}>
					 <Heading type="h2">{title}</Heading>
					</div>
					<Search searchBtnText={searchBtnText}/>
					<TopCategories topCategories={topCategoriesData}/>
					</div>
				</Grid>
	 		</div>
		</React.Fragment>
	 )
}

/** Props **/
SearchSection.defaultProps = {
	title: '',
	searchBtnText: 'search',
	topCategoriesData: []
}

SearchSection.propTypes = {
	title: PropTypes.string,
	searchBtnText: PropTypes.string,
	topCategoriesData: PropTypes.array
}

export default SearchSection;
