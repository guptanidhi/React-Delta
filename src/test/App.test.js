import React from "react";
import { mount } from "enzyme";
import App from "../App";
import CustomNavbar from '../components/Navbar';
import SearchSection from '../components/SearchSection';
import CustomSlider from '../components/Slider';

describe('App Component', () => {
	let wrapper;

	beforeAll(() => {
		wrapper = mount(<App />);
	})

	it("renders", () => {		
		expect(wrapper).toBeTruthy();
	});
	
	it("renders Navbar component", () => {
		const navBar = wrapper.find(CustomNavbar)
		expect(navBar).toBeTruthy();
	});
	
	it("renders Search Section component", () => {
		const searchSection = wrapper.find(SearchSection)
		expect(searchSection).toBeTruthy();
	});

	it("renders CustomSlider component", () => {
		const customSlider = wrapper.find(CustomSlider)
		expect(customSlider).toBeTruthy();
	});
	
	it("snapshot Testing", () => {
		const component = mount(<App />);
		expect(component).toMatchSnapshot();
	});
})
