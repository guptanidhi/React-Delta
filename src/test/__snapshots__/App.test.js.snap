// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot Testing 1`] = `ReactWrapper {}`;

exports[`App Component snapshot Testing 1`] = `ReactWrapper {}`;

exports[`Snapshot Testing 1`] = `ReactWrapper {}`;
